modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/security/permission/resource/CommerceOrderModelResourcePermissionLogic.java

```diff
@@ -148,6 +148,10 @@ public Boolean contains(
 				CommerceOrderActionKeys.
 					MANAGE_COMMERCE_ORDER_RESTRICTED_NOTES)) {
 
+			if (_hasRoleAccountSupplier(permissionChecker, commerceOrder)) {
```

Comment:
why this one?
supplier should already have the MANAGE_COMMERCE_ORDER_RESTRICTED_NOTES permission, we should only check that we are checking for an order the supplier has access to
I don't think this is right

Reply:
Depending on the context, the peformed check upon calling the `CommerceOrderNoteService` (`getCommerceOrderNotes`, `addCommerceOrderNote`, etc.) is solely based on the specific `MANAGE_COMMERCE_ORDER_NOTES` or `MANAGE_COMMERCE_ORDER_RESTRICTED_NOTES`.

This in turn checks the fact that the user belongs to accounts which contain `MANAGE_COMMERCE_ORDERS` (or being the owner of the note), and Supplier does not. The `_hasRoleAccountSupplier` in fact checks the allowance over orders on the channel the supplier is bound to, and wasn't covered in the case of notes, both for MVC actions and for POST/PATCH from `headless-commerce-admin-order`. We can take this on an offline call to make sure.

Reply:
@luca-pellizzon Changed with the completed logic and moved to `containsManageNotes`, could you please check? :)